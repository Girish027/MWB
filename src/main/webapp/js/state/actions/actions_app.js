
import _ from 'lodash';
import getUrl, { pathKey } from 'utils/apiUrls';
import { fetchUtil } from 'utils/fetchUtils';
import goToRoute from 'utils/routeHelpers';
import Constants from 'constants/Constants';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { logAmplitudeEvent } from 'utils/amplitudeUtils';
import axios from 'axios';
import api from 'utils/api';
import { setDefaultsForAPI } from 'utils/api';
import * as types from './types';

// Note to developers:
// Both Warning and Good Messages are shown for the period of notificationInterval.
// Bad, Persistant - till user closes the notification. This should only be used sparingly.
export const callLogIngestAPI = (message, logLevel) => {
  const promise = new Promise((resolve, reject) => {
    const url = getUrl(pathKey.logIngest, { logLevel });
    return new Promise((resolve, reject) => {
      api.post({
        url,
        headers: {
          'Content-Type': 'text/plain',
        },
        data: message,
      });
    });
  });
};

export const displayBadRequestMessage = (message) => (dispatch) => {
  dispatch({
    type: types.BAD_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_ERROR);
};

export const displayWarningRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.WARNING_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_WARNING);

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayGoodRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.GOOD_REQUEST,
    value: message,
  });

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayPersistantMessage = message => ({
  type: types.PERSISTENT_REQUEST,
  value: message,
});

export const stopShowingServerMessage = () => ({
  type: types.APP_STOP_SHOWING_SERVER_MESSAGE,
});

export const updateLoginSuccess = csrfToken => ({
  type: types.LOGIN_SUCCESS,
  value: csrfToken,
});

export const updateLoginInfo = (userId, userName, userDetails) => ({
  type: types.LOGIN_INFO,
  value: {
    userId,
    userName,
    userDetails,
  },
});

export const logoutFromOktaSuccess = () => ({
  type: types.LOGOUT_OKTA,
});

export const logoutFromMWBSuccess = () => ({
  type: types.LOGOUT_MWB,
});

export const contextMenuChange = contextMenuState => ({
  type: types.CONTEXT_MENU_CHANGE,
  contextMenuState,
});

export const modalDialogChange = modalDialogState => ({
  type: types.MODAL_DIALOG_CHANGE,
  modalDialogState,
});

const actionFileRemoveRequest = ({ fileId }) => ({
  type: types.FILE_REMOVE_REQUEST,
  fileId,
});

const actionFileRemoveError = ({ fileId, error }) => ({
  type: types.FILE_REMOVE_ERROR,
  fileId,
  error,
});

const actionFileRemoveSuccess = ({ fileId }) => ({
  type: types.FILE_REMOVE_SUCCESS,
  fileId,
});

export const updateUserGroups = data => ({
  type: types.UPDATE_USER_GROUPS,
  data,
});

const onFetchFailureRemoveFile = (error, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveError({ fileId, error }));
};

const onFetchSuccessRemoveFile = (data, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveSuccess({ fileId }));
};

export const removeFile = ({ fileId }) => (dispatch, getState) => {
  dispatch(actionFileRemoveRequest({ fileId }));

  const state = getState();
  const fetchUrl = getUrl(pathKey.files, { fileId, clientId: state.header.client.id });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'delete',
    onFetchSuccess: onFetchSuccessRemoveFile,
    onFetchFailure: onFetchFailureRemoveFile,
    dispatch,
    getState,
    params: {
      fileId,
    },
  });
};

const onFetchSuccessGetUserGroups = data => (dispatch) => {
  dispatch(updateUserGroups(data));
};

export const getUserGroups = ({ csrfToken }) => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.userGroups, { csrfToken });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetUserGroups,
    dispatch,
    getState,
  });
};

export const onFetchSuccessGetCSRFToken = data => (dispatch, getState) => {
  const csrfToken = data.value;
  setDefaultsForAPI(csrfToken);
  dispatch(updateLoginSuccess(csrfToken));
  dispatch(getUserGroups({ csrfToken }));
};

export const getCSRFToken = userId => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.csrfToken);

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetCSRFToken,
    dispatch,
    getState,
    params: {
      userId,
    },
  });
};

const logoutFromMWB = () => (dispatch, getState) => {
  dispatch(logoutFromOktaSuccess());
  dispatch(logoutFromMWBSuccess());
};

export const logoutFromOkta = () => (dispatch, getState) => {
  const state = getState();
  const { oAuthLogoutURL } = state.app;
  if (oAuthLogoutURL) {
    fetch(oAuthLogoutURL, {
      method: 'DELETE',
      credentials: 'include',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    })
      .then((response) => {
        window.location.href = Constants.LOGOUT;
        switch (response.status) {
        case 204:
          logoutFromMWB();
          logAmplitudeEvent(AmplitudeConstants.LOGOUT_EVENT, state);
          break;
        case 404:
        default:
          // TODO send log for these scenarios
          break;
        }
      })
      .catch((error) => {
        // TODO send log for these scenarios
      });
  }
};

export const changeRoute = (routeName, data, history) => (dispatch, getState) => {
  const state = getState();
  const routeData = Object.assign({}, {
    client: !_.isNil(state.header.client) ? state.header.client : {
      id: 0,
      name: 'Loading...',
    },
    projectId: state.selectedProjectId,
  }, data);
  goToRoute(routeName, routeData, history);
};
