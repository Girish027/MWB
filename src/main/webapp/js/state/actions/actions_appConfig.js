import 'regenerator-runtime/runtime';

import getUrl, { pathKey } from 'utils/apiUrls';

import { fetchUtil } from 'utils/fetchUtils';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { initializeAnalytics, logAmplitudeEvent } from 'utils/amplitudeUtils';
import api from 'utils/api';
import * as types from './types';

const appConfigSuccess = data => ({
  type: types.APP_CONFIG_SUCCESS,
  data,
});

const appConfigFailure = data => ({
  type: types.APP_CONFIG_FAILURE,
  data,
});

const appConfigRequest = () => ({
  type: types.APP_CONFIG_REQUEST,
});

export const onAppConfigSuccess = data => (dispatch, getState) => {
  dispatch(appConfigSuccess(data));
  const state = getState();
  initializeAnalytics(state);
  logAmplitudeEvent(AmplitudeConstants.LOGIN_EVENT, state);
};

export const onAppConfigFailure = error => (dispatch) => {
  dispatch(appConfigFailure(error));
};

export const fetchAppConfig = () => (dispatch, getState) => {
  dispatch(appConfigRequest());
  const fetchUrl = getUrl(pathKey.appConfig);

  api.get({
    url: fetchUrl,
    onApiSuccess: onAppConfigSuccess,
    onApiError: onAppConfigFailure,
    dispatch,
    getState,
  });
};
