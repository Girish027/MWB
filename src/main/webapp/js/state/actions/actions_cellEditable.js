import * as types from './types';

export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_CREATE,
  stateKey,
});

export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_REMOVE,
  stateKey,
});

export const setActiveCell = ({ stateKey, activeRowIndex, activeColumnIndex }) => ({
  type: types.CELL_EDITABLE_SET_ACTIVE_CELL,
  stateKey,
  activeRowIndex,
  activeColumnIndex,
});

export const updateValue = ({ stateKey, value }) => ({
  type: types.CELL_EDITABLE_UPDATE_VALUE,
  stateKey,
  value,
});
