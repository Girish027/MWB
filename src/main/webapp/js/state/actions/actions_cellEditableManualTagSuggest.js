import store from 'state/configureStore';
import errorMessageUtil from 'utils/ErrorMessageUtil';
import getUrl, { pathKey } from 'utils/apiUrls';
import * as types from './types';

let controller;
export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_CREATE,
  stateKey,
});
export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_REMOVE,
  stateKey,
});

const getSuggestedTagsRequest = ({ stateKey, projectId, intent }) => ({
  type: types.GET_SUGGESTED_TAGS_REQUEST,
  stateKey,
  projectId,
  intent,
});

const getSuggestedTagsError = ({
  stateKey, projectId, intent, error,
}) => ({
  type: types.GET_SUGGESTED_TAGS_ERROR,
  stateKey,
  projectId,
  intent,
  error,
});

const getSuggestedTagsSuccess = ({
  stateKey, projectId, intent, suggestedTags,
}) => ({
  type: types.GET_SUGGESTED_TAGS_SUCCESS,
  stateKey,
  projectId,
  intent,
  suggestedTags,
});

export const getSuggestedTags = ({ stateKey, projectId, intent }) => (dispatch, getState) => new Promise((resolve, reject) => {
  if (controller) {
    controller.abort();
  }
  controller = new AbortController();
  const signal = controller.signal;
  dispatch(getSuggestedTagsRequest({ stateKey, projectId, intent }));
  const state = getState();
  const { csrfToken } = state.app;
  const clientId = state.header.client.id;

  const fetchUrl = getUrl(pathKey.suggestedTags, { projectId, intent, clientId });

  fetch(fetchUrl, {
    signal,
    credentials: 'same-origin',
    headers: {
      'X-CSRF-TOKEN': csrfToken,
    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((json) => {
      const suggestedTags = json && json.length ? json : [];
      const intentIndex = suggestedTags.indexOf(intent);
      if (intentIndex >= 0) {
        suggestedTags.slice(intentIndex, 1);
      }
      dispatch(getSuggestedTagsSuccess({
        stateKey, projectId, intent, suggestedTags,
      }));
      resolve([...suggestedTags]);
    })
    .catch((error) => {
      if (error.name === 'AbortError') { return; }
      dispatch(getSuggestedTagsError({
        stateKey, projectId, intent, error,
      }));
      reject(error);
    });
});

export const cursorDown = ({ stateKey }) => ({
  type: types.CELL_SUGGEST_CURSOR_DOWN,
  stateKey,
});

export const cursorUp = ({ stateKey }) => ({
  type: types.CELL_SUGGEST_CURSOR_UP,
  stateKey,
});
