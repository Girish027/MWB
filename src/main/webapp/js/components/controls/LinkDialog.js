import React, { Component } from 'react';
import { Dialog, TextField } from '@tfs/ui-components';
import PropTypes from 'prop-types';
import * as appActions from 'state/actions/actions_app';
import _ from 'lodash';
import Constants from 'constants/Constants';
import { headerIcon } from 'styles';

export default class LinkDialog extends Component {
  constructor(props) {
    super(props);
    this.props = props;
    this.onClickCancel = this.onClickCancel.bind(this);
    this.onClickOk = this.onClickOk.bind(this);
    this.onChange = this.onChange.bind(this);
    this.onKeyPress = this.onKeyPress.bind(this);

    this.state = {
      linkTextField: '',
    };

    this.styleOverride = {
      childContainer: {
        marginTop: '10px',
        marginBottom: '10px',
      },
      cancel: {
        backgroundColor: 'unset',
        paddingLeft: '10px',
        paddingRight: '10px',
      },
      ok: {
        marginLeft: '10px',
        paddingLeft: '25px',
        paddingRight: '25px',
      },
      ...headerIcon,
    };
  }

  onChange(event) {
    event.preventDefault();
    const { value } = event.target;
    this.setState({
      linkTextField: value,
    });
  }

  onKeyPress(event) {
    // TODO : write KeyboardUtils APIs
    if (event.key === 'Enter'
        || event.key === 'Tab') {
      event.preventDefault();
      this.onClickOk();
    }
  }

  onClickCancel() {
    const { dispatch, onClickCancel } = this.props;
    onClickCancel();
    dispatch(appActions.modalDialogChange(null));
  }

  onClickOk() {
    const { dispatch, onClickRunTest } = this.props;
    const { linkTextField } = this.state;
    dispatch(appActions.modalDialogChange({
      dispatch,
      type: Constants.DIALOGS.PROGRESS_DIALOG,
      message: Constants.TEST_IN_PROGRESS,
    }));
    onClickRunTest(linkTextField);
  }

  render() {
    const { header } = this.props;
    const { linkTextField } = this.state;

    return (
      <div>
        <Dialog
          size="small"
          headerChildren={header}
          onClickCancel={this.onClickCancel}
          onClickOk={this.onClickOk}
          onClickClose={this.onClickCancel}
          isOpen
          okDisabled={_.isNil(linkTextField) || linkTextField.length === 0}
          okChildren={Constants.RUN_TEST}
          styleOverride={this.styleOverride}
        >
          <div style={{
            textAlign: 'left',
            paddingTop: '40px',
          }}
          >
            <TextField
              type="text"
              name="link"
              placeholder="Enter URL"
              onChange={this.onChange}
              onKeyDown={this.onKeyPress}
              style={{ width: '435px' }}
            />
          </div>
        </Dialog>
      </div>
    );
  }
}

LinkDialog.propTypes = {
  header: PropTypes.string.isRequired,
  onClickCancel: PropTypes.func,
  onClickRunTest: PropTypes.func.isRequired,
  dispatch: PropTypes.func,
};

LinkDialog.defaultProps = {
  onClickCancel: () => {},
};
