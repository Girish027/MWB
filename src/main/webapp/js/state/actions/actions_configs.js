import fetch from 'isomorphic-fetch';

import errorMessageUtil from 'utils/ErrorMessageUtil';
import getUrl, { pathKey } from 'utils/apiUrls';
import { submitModel } from 'state/actions/actions_models';
import * as apiUtils from 'utils/apiUtils';
import { getLanguage } from 'state/constants/getLanguage';
import Constants from 'constants/Constants';
import api from 'utils/api';
import * as appActions from './actions_app';
import * as types from './types';

const languageConstants = getLanguage();
const { DISPLAY_MESSAGES } = languageConstants;

export const configUpload = () => ({
  type: types.CONFIG_UPLOAD,
});

export const configUploadSuccess = () => ({
  type: types.CONFIG_UPLOAD_SUCCESS,
});

export const configUploadFail = () => ({
  type: types.CONFIG_UPLOAD_FAIL,
});

export const configCreated = ({ config }) => ({
  type: types.CONFIG_CREATED,
  config,
});

export const configUpdated = ({ config }) => ({
  type: types.CONFIG_UPDATED,
  config,
});

export const configEditUpdate = config => ({
  type: types.CONFIG_EDIT_UPDATE,
  config,
});

export const convertToSpeechConfig = () => ({
  type: types.CONVERT_TO_SPEECH_CONFIG,
});

export const convertToDigitalConfig = () => ({
  type: types.CONVERT_TO_DIGITAL_CONFIG,
});

export const updateTrainingConfigValidity = isTrainingConfigsValid => ({
  type: types.UPDATE_TRAINING_CONFIG_VALIDITY,
  isTrainingConfigsValid,
});

export const updateTransformationValidity = isTransformationValid => ({
  type: types.UPDATE_TRANSFORMATION_VALIDITY,
  isTransformationValid,
});

export const createModelConfig = (data, projectId, csrfToken, clientId) => (dispatch, getState) => new Promise((resolve, reject) => {
  const configData = {
    name: data.name,
    description: data.description,
    projectId,
    configFile: JSON.stringify(data, null, 2),
  };

  const bodyData = JSON.stringify(configData, null, 2);

  const fetchUrl = getUrl(pathKey.config, { clientId });

  fetch(fetchUrl, {
    method: 'POST',
    credentials: 'same-origin',
    headers: {
      'X-CSRF-TOKEN': csrfToken,
      'Content-Type': 'application/json',
      Accept: 'application/json',
    },
    body: bodyData,
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((response) => {
      const newConfig = apiUtils.normalizeIds(response);
      newConfig._key = newConfig.id;
      newConfig.status = 'NULL';
      dispatch(configCreated({ config: newConfig }));
      dispatch(fetchConfigById(newConfig.id));

      dispatch(appActions.displayGoodRequestMessage(DISPLAY_MESSAGES.createdSuccess(newConfig.name)));
    })
    .catch((error) => {
      switch (error.status) {
      case 400: {
        if (error.response.body.errors) {
          error.response.body.errors.forEach((e) => {
            const fieldName = e.field.substring(e.field.lastIndexOf('.') + 1, e.field.length);
            data[fieldName] = `${fieldName} required!`;
          });
          dispatch(appActions.displayBadRequestMessage(data));
        }
        const messageLoggedIn = DISPLAY_MESSAGES.createFailed;
        dispatch(appActions.displayBadRequestMessage(messageLoggedIn));
        break;
      }

      case 409: {
        const messageAlreadyExists = DISPLAY_MESSAGES.configAlreadyExists;
        dispatch(appActions.displayBadRequestMessage(messageAlreadyExists));
        break;
      }

      default:
        errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage);
      }
    });
});

export const onSubmitConfigAndModelSuccess = (modelTypeToBeCreated, data, clientId, response = {}) => (dispatch, getState) => {
  const { model, wordclassFile } = data;
  const newConfig = apiUtils.normalizeIds(response);
  response._key = newConfig.id;
  response.status = 'NULL';
  dispatch(configCreated({ config: newConfig }));
  model.configId = newConfig.id;
  model.modelType = modelTypeToBeCreated;
  model.speechConfigId = null;

  if (modelTypeToBeCreated === Constants.DIGITAL_SPEECH_MODEL && JSON.stringify(wordclassFile) !== '{}') {
    dispatch(importSpeechFile(data, clientId));
  } else {
    dispatch(submitModel(model));
  }

  dispatch(appActions.displayGoodRequestMessage(DISPLAY_MESSAGES.createdSuccess(newConfig.name)));
};

export const onSubmitConfigAndModelFailure = (error = {}) => (dispatch, getState) => {
  const { status } = error;
  if (status == 400) {
    const data = {};
    if (error.response.body.errors) {
      error.response.body.errors.forEach((e) => {
        const fieldName = e.field.substring(e.field.lastIndexOf('.') + 1, e.field.length);
        data[fieldName] = `${fieldName} required!`;
      });
      dispatch(appActions.displayBadRequestMessage(data));
    }
    const messageLoggedIn = DISPLAY_MESSAGES.createFailed;
    dispatch(appActions.displayBadRequestMessage(messageLoggedIn));
  } else if (status == 409) {
    const messageAlreadyExists = DISPLAY_MESSAGES.configAlreadyExists;
    dispatch(appActions.displayBadRequestMessage(messageAlreadyExists));
  } else {
    errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage);
  }
};


export const submitConfigAndModel = (data, modelTypeToBeCreated = Constants.DIGITAL_MODEL) => (dispatch, getState) => {
  const configName = `${data.model.name}-cfg`;
  data.config.name = configName;
  const { model } = data;
  const configData = {
    name: configName,
    description: model.description,
    projectId: model.projectId,
    configFile: JSON.stringify(data.config, null, 2),
  };
  const state = getState();
  const clientId = state.header.client.id;
  const url = getUrl(pathKey.config, { clientId });
  return new Promise((resolve, reject) => {
    api.post({
      dispatch,
      getState,
      url,
      data: configData,
      onApiSuccess: response => onSubmitConfigAndModelSuccess(modelTypeToBeCreated, data, clientId, response),
      onApiError: onSubmitConfigAndModelFailure,
    });
  });
};

export const submitConfigAndModelForSpeech = (data, modelTypeToBeCreated = Constants.SPEECH_MODEL) => (dispatch, getState) => {
  const configName = `${data.model.name}-cfg`;
  const { model } = data;
  const wordclassFile = data.model.wordClassFile;
  const formDataParams = new FormData();
  formDataParams.append('name', configName);
  formDataParams.append('description', '');
  formDataParams.append('projectId', model.projectId);
  formDataParams.append('file', wordclassFile, wordclassFile.name);
  const state = getState();
  const clientId = state.header.client.id;
  const url = getUrl(pathKey.config, { clientId });
  fetch(url, {
    method: 'POST',
    credentials: 'same-origin',
    body: formDataParams,
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((response) => {
      dispatch(addSpeechConfig(response, data));
    })
    .catch((error) => {
      onSubmitConfigAndModelFailure(error);
    });
};

export const addSpeechConfig = (response, data) => (dispatch, getState) => {
  const { model } = data;
  const newConfig = apiUtils.normalizeIds(response);
  response._key = newConfig.id;
  model.speechConfigId = newConfig.id;
  dispatch(submitModel(model));
};

export const addSpeechWithWordClassFile = (data, clientId) => (dispatch, getState) => {
  const { model, wordclassFile = {} } = data;
  const configName = `${data.model.name}-cfg`;

  const formDataParams = new FormData();
  formDataParams.append('name', configName);
  formDataParams.append('description', '');
  formDataParams.append('projectId', model.projectId);
  formDataParams.append('file', wordclassFile, wordclassFile.name);

  const fetchUrl = getUrl(pathKey.config, { clientId });
  fetch(fetchUrl, {
    method: 'POST',
    credentials: 'same-origin',
    body: formDataParams,
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((response) => {
      dispatch(addSpeechConfig(response, data));
    })
    .catch((error) => {
      dispatch(appActions.displayBadRequestMessage(DISPLAY_MESSAGES.fileImportFail));
    });
};

export const addSpeechWithoutWordClassFile = (data, clientId) => (dispatch, getState) => {
  const { model } = data;
  const { configId } = model;
  const fetchUrl = getUrl(pathKey.getSpeechConfig, { clientId, configId });
  fetch(fetchUrl, {
    method: 'GET',
    credentials: 'same-origin',
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((response) => {
      dispatch(addSpeechConfig(response, data));
    })
    .catch((error) => {
      dispatch(appActions.displayBadRequestMessage(DISPLAY_MESSAGES.fileImportFail));
    });
};

export const importSpeechFile = (data, clientId) => (dispatch, getState) => {
  data = {
    ...data,
    clientId,
  };
  const { wordclassFile = {} } = data;
  if (JSON.stringify(wordclassFile) !== '{}') {
    dispatch(addSpeechWithWordClassFile(data, clientId));
  } else {
    dispatch(addSpeechWithoutWordClassFile(data, clientId));
  }
};

export const requestConfig = configId => ({
  type: types.REQUEST_CONFIG,
  projectId: configId,
});

export const receiveConfig = config => ({
  type: types.RECEIVE_CONFIG,
  config,
  receivedAt: Date.now(),
});

export const fetchConfigById = configId => (dispatch, getState) => {
  dispatch(requestConfig(configId));
  const state = getState();
  const { csrfToken } = state.app;
  const clientId = state.header.client.id;

  const fetchUrl = getUrl(pathKey.configById, { configId, clientId });
  return fetch(fetchUrl, {
    credentials: 'same-origin',
    headers: {
      'X-CSRF-TOKEN': csrfToken,

    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((json) => {
      let config = null;
      try {
        config = JSON.parse(json.configFile);
      } catch (error) {
        errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage);
      }
      dispatch(receiveConfig(config));
    })
    .catch(error => errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage));
};

export const showTransformationDeleteDialog = show => ({
  type: types.CONFIG_SHOW_TRANSFORMATION_DELETE_DIALOG,
  show,
});

export const showTransformationPredefinedDialog = show => ({
  type: types.CONFIG_SHOW_TRANSFORMATION_PREDEFINED_DIALOG,
  show,
});

export const showTransformationAddDialog = show => ({
  type: types.CONFIG_SHOW_TRANSFORMATION_ADD_DIALOG,
  show,
});
