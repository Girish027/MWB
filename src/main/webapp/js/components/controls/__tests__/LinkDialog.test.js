import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as actionsApp from 'state/actions/actions_app';
import Constants from 'constants/Constants';
import LinkDialog from 'components/controls/LinkDialog';

describe('<LinkDialog />', () => {
  let wrapper;
  const props = {
    header: 'Link Dialog',
    onCancel: () => {},
    dispatch: () => {},
    onClickRunTest: () => {},
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  describe('Creating an instance with test props', () => {
    beforeEach(() => {
      wrapper = shallow(<LinkDialog
        {...props}
      />);
    });

    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });

    test('Snapshots should render correctly', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Function Call Handler:', () => {
    beforeAll(() => {
      props.header = '',
      props.dispatch = jest.fn();
      props.onOk = jest.fn();
      props.onCancel = jest.fn();
      props.onClickRunTest = jest.fn();
      actionsApp.modalDialogChange = jest.fn(() => 'called modalDialogChange');
    });

    beforeEach(() => {
      wrapper = shallow(<LinkDialog
        {...props}
      />);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    describe('onClickCancel:', () => {
      test('should onClickCancel call click handler', () => {
        wrapper.instance().onClickCancel();
        expect(props.onCancel).toHaveBeenCalled;
        expect(props.dispatch).toHaveBeenCalledWith('called modalDialogChange');
      });
    });

    describe('onClickOk:', () => {
      test('should show the Progress Dialog', () => {
        wrapper.setState({
          linkTextField: '/something/text.wav',
        });
        wrapper.instance().onClickOk();
        expect(actionsApp.modalDialogChange).toHaveBeenCalledWith({
          dispatch: props.dispatch,
          type: Constants.DIALOGS.PROGRESS_DIALOG,
          message: Constants.TEST_IN_PROGRESS,
        });
        expect(props.dispatch).toHaveBeenCalledWith('called modalDialogChange');
      });

      test('should run test for the given link', () => {
        wrapper.setState({
          linkTextField: '/something/text.wav',
        });
        wrapper.instance().onClickOk();
        expect(props.onClickRunTest).toHaveBeenCalledWith(wrapper.state().linkTextField);
      });
    });

    describe('onChange:', () => {
      test('should prevent default behaviour of triggered event and update state', () => {
        const event = {
          preventDefault: jest.fn(),
          target: {
            value: '/something/text.wav',
          },
        };
        wrapper.instance().onChange(event);
        expect(event.preventDefault).toHaveBeenCalled;
        expect(wrapper.state().linkTextField).toEqual(event.target.value);
      });
    });

    describe('onKeyPress:', () => {
      test('should prevent default behaviour of triggered event for Enter key and call click handler', () => {
        const event = {
          preventDefault: jest.fn(),
          key: 'Enter',
        };
        wrapper.instance().onClickOk = jest.fn();
        wrapper.instance().onKeyPress(event);
        expect(event.preventDefault).toHaveBeenCalled;
        expect(wrapper.instance().onClickOk).toHaveBeenCalled;
      });

      test('should prevent default behaviour of triggered event for Enter key and call click handler', () => {
        const event = {
          preventDefault: jest.fn(),
          key: 'Tab',
        };
        wrapper.instance().onClickOk = jest.fn();
        wrapper.instance().onKeyPress(event);
        expect(event.preventDefault).toHaveBeenCalled;
        expect(wrapper.instance().onClickOk).toHaveBeenCalled;
      });

      test('should not handle event if any other key is pressed', () => {
        const event = {
          preventDefault: jest.fn(),
          key: 'A',

        };
        wrapper.instance().onClickOk = jest.fn();
        wrapper.instance().onKeyPress(event);
        expect(event.preventDefault).not.toHaveBeenCalled;
        expect(wrapper.instance().onClickOk).not.toHaveBeenCalled;
      });
    });
  });
});
