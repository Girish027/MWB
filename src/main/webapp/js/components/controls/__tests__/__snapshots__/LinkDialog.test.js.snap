// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkDialog /> Creating an instance with test props Snapshots should render correctly 1`] = `
<div>
  <withTheme(DialogComponent)
    headerChildren="Link Dialog"
    isOpen={true}
    okChildren="RUN TEST"
    okDisabled={true}
    onClickCancel={[Function]}
    onClickClose={[Function]}
    onClickOk={[Function]}
    size="small"
    styleOverride={
      Object {
        "cancel": Object {
          "backgroundColor": "unset",
          "paddingLeft": "10px",
          "paddingRight": "10px",
        },
        "childContainer": Object {
          "marginBottom": "10px",
          "marginTop": "10px",
        },
        "header": Object {
          "borderBottom": "1px solid",
          "borderBottomColor": "#DADADA",
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "center",
          "title": Object {
            "fontSize": "20px",
            "fontWeight": "bold",
            "marginLeft": "23px",
            "marginRight": "20px",
          },
          "titleIcon": Object {
            "marginRight": "20px",
          },
          "width": "100%",
        },
        "ok": Object {
          "marginLeft": "10px",
          "paddingLeft": "25px",
          "paddingRight": "25px",
        },
      }
    }
  >
    <div
      style={
        Object {
          "paddingTop": "40px",
          "textAlign": "left",
        }
      }
    >
      <withTheme(TextField)
        name="link"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Enter URL"
        style={
          Object {
            "width": "435px",
          }
        }
        type="text"
      />
    </div>
  </withTheme(DialogComponent)>
</div>
`;
